src/AndroidManifest.js

```diff
@@ -0,0 +1,769 @@
+
+import * as util from './util';
+
+const androidAttrPrefixRegExp = /^android\:/;
+const defaultDOMParserArgs = { errorHandler: function(){} };
+
+const tags = {
+	// according to http://developer.android.com/guide/topics/manifest/meta-data-element.html,
+	// <meta-data> is not intended to be a direct descendent of <application> but it is required
+	// that we support it so that we can pass a Google Maps API key to their library
+	'application': /^(activity|activity-alias|meta-data|provider|receiver|service|uses\-library)$/
+};
+
+const tagAttrs = {
+	'application': /^(allowTaskReparenting|allowBackup|backupAgent|debuggable|description|enabled|hasCode|hardwareAccelerated|icon|killAfterRestore|largeHeap|label|logo|manageSpaceActivity|name|permission|persistent|process|restoreAnyVersion|requiredAccountType|restrictedAccountType|supportsRtl|taskAffinity|testOnly|theme|uiOptions|vmSafeMode)$/,
+	'activity': /^(allowTaskReparenting|alwaysRetainTaskState|clearTaskOnLaunch|configChanges|enabled|excludeFromRecents|exported|finishOnTaskLaunch|hardwareAccelerated|icon|label|launchMode|multiprocess|name|noHistory|parentActivityName|permission|process|screenOrientation|stateNotNeeded|taskAffinity|theme|uiOptions|windowSoftInputMode)$/,
+	'activity-alias': /^(enabled|exported|icon|label|name|permission|targetActivity)$/,
+	'data': /^(host|mimeType|path|pathPattern|pathPrefix|port|scheme)$/,
+	'intent-filter': /^(icon|label|priority)$/,
+	'meta-data': /^(name|resource|value)$/,
+	'path-permission': /^(path|pathPrefix|pathPattern|permission|readPermissions|writePermissions)$/,
+	'provider': /^(authorities|enabled|exported|grantUriPermissions|icon|initOrder|label|multiprocess|name|permission|process|readPermission|syncable|writePermission)$/,
+	'receiver': /^(enabled|exported|icon|label|name|permission|process)$/,
+	'service': /^(enabled|exported|icon|isolatedProcess|label|name|permission|process)$/,
+	'uses-library': /^(name|required)$/,
+	'uses-sdk': /^(name|required)$/
+};
+
+
+/**
+ * Provides methods to process the AndroidManifest.xml file.
+ *
+ * @class
+ * @constructor
+ */
+export default class AndroidManifest {
+	/**
+	 * Creates an Android Manifest object.
+	 *
+	 * @param {String} file - The path of the AndroidManifest.xml file.
+	 */
+	constructor(file) {
+		file = util.expandPath(file);
+		if (!util.existsSync(file)) {
+			throw new Error('AndroidManifest.xml file does not exist.');
+		}
+
+		this.load(file);
+	}
+
+	/**
+	 * Parses Android Manifest from file.
+	 *
+	 * @param {String} file - The path of the AndroidManifest.xml file.
+	 * @returns {AndroidManifest}
+	 * @access public
+	 */
+	load(file) {
+		if (typeof file !== 'string' || !file) {
+			throw new TypeError('Expected file to be a string.');
+		}
+
+		if (!fs.existsSync(file)) {
+			throw new Error('AndroidManifest.xml file does not exist.');
+		}
+
+		this.toJS((new DOMParser(defaultDOMParserArgs).parseFromString(fs.readFileSync(file).toString(), 'text/xml')).documentElement);
```

Comment:
Read the file in and pass the contents to `this.parse()` instead of duplicating the `DOMParser` and `toJS()` code.